HostChecker test failure

**Do you want to request a *feature* or report a *bug*?**
bug
**What is the current behavior?**
It happened to be only 1 time, but I was able to record backtrace:

```
=== RUN   TestHostChecker
panic: sync: negative WaitGroup counter

goroutine 9681 [running]:
sync.(*WaitGroup).Add(0xc431996910, 0xffffffffffffffff)
	/usr/local/Cellar/go/1.8.1/libexec/src/sync/waitgroup.go:75 +0x1c9
sync.(*WaitGroup).Done(0xc431996910)
	/usr/local/Cellar/go/1.8.1/libexec/src/sync/waitgroup.go:100 +0x34
github.com/TykTechnologies/tyk.TestHostChecker.func1(0x19b9c9e, 0x8, 0x18f90a0, 0xc430ca7e80, 0xc4293433b0, 0x27)
	/Users/buger/Documents/work/go/src/github.com/TykTechnologies/tyk/host_checker_test.go:85 +0x2a
github.com/TykTechnologies/tyk.(*testEventHandler).HandleEvent(0xc423fd2ba0, 0x19b9c9e, 0x8, 0x18f90a0, 0xc430ca7e80, 0xc4293433b0, 0x27)
	/Users/buger/Documents/work/go/src/github.com/TykTechnologies/tyk/host_checker_test.go:58 +0x45
created by github.com/TykTechnologies/tyk.fireEvent
	/Users/buger/Documents/work/go/src/github.com/TykTechnologies/tyk/event_system.go:173 +0x1c3
exit status 2
FAIL	github.com/TykTechnologies/tyk	3.448s
```

**Which versions of Tyk affected by this issue? Did this work in previous versions of Tyk?**
master

--- a/host_checker.go
+++ b/host_checker.go
@@ -76,6 +76,9 @@ func (h *HostUptimeChecker) getStaggeredTime() time.Duration {
 
 func (h *HostUptimeChecker) HostCheckLoop() {
 	for !h.stopLoop {
+		if runningTests {
+			<-hostCheckTicker
+		}
 		h.resetListMu.Lock()
 		if h.doResetList {
 			if h.newList != nil {
@@ -93,9 +96,7 @@ func (h *HostUptimeChecker) HostCheckLoop() {
 			}
 		}
 
-		if runningTests {
-			<-hostCheckTicker
-		} else {
+		if !runningTests {
 			time.Sleep(h.getStaggeredTime())
 		}
 	}

